test/no-hex-escape.js

```diff
@@ -0,0 +1,75 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-hex-escape';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const error = {
+	ruleId: 'no-hex-escape',
+	message: 'Use unicode escape codes instead of hexadecimal escapes.'
+};
+
+ruleTester.run('no-hex-escape', rule, {
+	valid: [
+		`const foo = 'foo'`,
+		`const foo = '\\u00b1'`,
+		`const foo = '\\u00b1\\u00b1'`,
+		`const foo = 'foo\\u00b1'`,
+		`const foo = 'foo\\u00b1foo'`,
+		`const foo = '\\u00b1foo'`,
+		'const foo = 42',
+		'const foo = `foo`',
+		'const foo = `42`'
+	],
+	invalid: [
+		{
+			code: `const foo = '\\xb1'`,
+			errors: [error],
+			output: `const foo = '\\u00b1'`
+		},
+		{
+			code: `const foo = '\\xb1\\xb1'`,
+			errors: [error],
+			output: `const foo = '\\u00b1\\u00b1'`
+		},
+		{
+			code: `const foo = '\\xb1foo'`,
+			errors: [error],
+			output: `const foo = '\\u00b1foo'`
+		},
+		{
+			code: `const foo = '\\xb1\\xb1'`,
+			errors: [error],
+			output: `const foo = '\\u00b1\\u00b1'`
+		},
+		{
+			code: `const foo = '\\xd8\\x3d\\xdc\\xa9'`,
+			errors: [error],
+			output: `const foo = '\\u00d8\\u003d\\u00dc\\u00a9'`
+		},
+		{
+			code: `const foo = 'foo\\xb1'`,
+			errors: [error],
+			output: `const foo = 'foo\\u00b1'`
+		},
+		{
+			code: `const foo = 'foo\\x12foo\\x34'`,
+			errors: [error],
+			output: `const foo = 'foo\\u0012foo\\u0034'`
+		},
+		{
+			code: `const foo = '42\\x1242\\x34'`,
+			errors: [error],
+			output: `const foo = '42\\u001242\\u0034'`
+		},
+		{
+			code: 'const foo = `\\xb1`',
```

Comment:
Add some more template literal tests, especially with multiple elements. Also add more passing tests like that.

See: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/63#discussion_r84426600